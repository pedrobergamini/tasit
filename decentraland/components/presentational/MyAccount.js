import React from "react";
import { StyleSheet, View } from "react-native";
import Colors from "@constants/Colors";
import PropTypes from "prop-types";
import { responsiveWidth } from "react-native-responsive-dimensions";
import MyAccountCreationStatusItem from "./MyAccountCreationStatusItem";
import ActionStatus from "@constants/ActionStatus";
import MyAccountProgress from "./MyAccountProgress";

export default function MyAccount({ progress, creationActions }) {
  return (
    <View style={styles.container}>
      <MyAccountProgress progress={progress} />
      <View style={styles.actionItemsContainer}>
        {creationActions.map(action => {
          const { name, status } = action;

          return (
            <MyAccountCreationStatusItem
              key={name}
              name={name}
              status={status}
            />
          );
        })}
      </View>
    </View>
  );
}

MyAccount.propTypes = {
  progress: PropTypes.number,
  creationActions: PropTypes.arrayOf(
    PropTypes.shape({
      status: PropTypes.oneOf(Object.values(ActionStatus)),
      name: PropTypes.string,
    })
  ),
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: Colors.backgroundColor,
    alignItems: "flex-start",
    justifyContent: "flex-start",
  },
  actionItemsContainer: {
    flex: 6,
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "center",
    paddingLeft: responsiveWidth(20),
  },
});
