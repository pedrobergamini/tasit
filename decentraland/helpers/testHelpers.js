import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () =>
    "0x1234567890123456789012345678901234567890123456789012345678901234",
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
};
