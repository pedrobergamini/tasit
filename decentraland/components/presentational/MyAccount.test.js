import React from "react";
import { shallow } from "enzyme";
import MyAccount from "./MyAccount";
import { accountCreationActions } from "@helpers/testHelpers";

describe("MyAccount", () => {
  it("renders the component", async () => {
    const progress = 0.5;
    expect(
      shallow(
        <MyAccount
          progress={progress}
          creationActions={accountCreationActions}
        />
      )
    ).toMatchSnapshot();
  });
});
