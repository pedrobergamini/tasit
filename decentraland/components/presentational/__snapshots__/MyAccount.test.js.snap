// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAccount renders the component 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <MyAccountProgress
    progress={0.5}
  />
  <Component
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 6,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingLeft": 150,
      }
    }
  >
    <MyAccountCreationStatusItem
      key="Account created"
      name="Account created"
      status="done"
    />
    <MyAccountCreationStatusItem
      key="ETH"
      name="ETH"
      status="pending"
    />
    <MyAccountCreationStatusItem
      key="MANA"
      name="MANA"
      status="missing"
    />
  </Component>
</Component>
`;
